x/group/keeper/proposal_executor.go

```diff
@@ -0,0 +1,39 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/module/server"
+	"github.com/cosmos/cosmos-sdk/x/group"
+)
+
+func (s msgServer) execMsgs(ctx context.Context, derivationKey []byte, proposal group.Proposal) error {
+	derivedKey := s.server.key.Derive(derivationKey)
+	msgs := proposal.GetMsgs()
+
+	for _, msg := range msgs {
+		var reply interface{}
+
+		// Execute the message using the derived key,
+		// this will verify that the message signer is the group account.
+		err := derivedKey.Invoke(ctx, server.TypeURL(msg), msg, reply)
```

Comment:
we shouldn't use ADR 033 here, have a look at the previous group implementation in https://github.com/regen-network/cosmos-modules/ but use msg service router instead of the legacy router https://github.com/regen-network/cosmos-modules/blob/master/incubator/group/proposal_executor.go